ui/prometheus-plugin/src/plugins/prometheus-time-series-query/PrometheusTimeSeriesQueryEditor.tsx

```diff
@@ -41,9 +43,18 @@ export function PrometheusTimeSeriesQueryEditor(props: PrometheusTimeSeriesQuery
   const datasourceSelectLabelID = `prom-datasource-label-${selectedDatasource.name || 'default'}`;
 
   const { data: client } = useDatasourceClient<PrometheusClient>(selectedDatasource);
-  const promURL = client?.options.datasourceUrl;
   const { data: datasourceResource } = useDatasource(selectedDatasource);
 
+  const promURL =
+    (datasourceResource?.plugin.spec as PrometheusDatasourceSpec).directUrl ?? client?.options.datasourceUrl;
```

Comment:
I am wondering if `client?.options.datasourceUrl` would be enough. I am pretty sure it's the case, but I can't find where it is set.

I have this feeling because I know based on the proxy configuration we are giving to the client, what URL it should use so `directUrl` or `plugin.spec.proxy.spec.url`

Reply:
@Gladorme or @celian-garcia `client?.options.datasourceUrl;` should be enough to get the accurate URL right ? Otherwise it wouldn't work today whether the proxy is used or not

Reply:
Hi, I just finished testing it. I think `client?.options.datasourceUrl;` should be enough and I've made the commit.